snuba/admin/auth.py

```diff
@@ -33,10 +37,41 @@ def authorize_request() -> AdminUser:
     return _set_roles(provider())
 
 
+def _is_member_of_group(user: AdminUser, group: str) -> bool:
+    # TODO this will be handled by gcp identity once we have the service account
+    # https://cloud.google.com/identity/docs/how-to/query-memberships#searching_for_all_memberships_in_a_group
+    return False
+
+
+def get_iam_roles_from_file(user: AdminUser) -> Sequence[str]:
+    iam_roles = []
+    try:
+        with open(settings.ADMIN_IAM_POLICY_FILE, "r") as policy_file:
+            policy = json.load(policy_file)
+            for binding in policy["bindings"]:
+                role: str = binding["role"].split("roles/")[-1]
+                for member in binding["members"]:
+                    if f"user:{user.email}" == member:
+                        iam_roles.append(role)
+                        break
+                    if member.startswith("group:"):
+                        group = member.split("group:")[-1]
+                        if _is_member_of_group(user, group):
+                            iam_roles.append(role)
+                            break
+    except FileNotFoundError:
+        logger.warn(
+            f"IAM policy file not found {settings.ADMIN_IAM_POLICY_FILE}. Using default roles only."
+        )
+
+    return iam_roles
+
+
 def _set_roles(user: AdminUser) -> AdminUser:
     # todo: depending on provider convert user email
     # to subset of DEFAULT_ROLES based on IAM roles
-    user.roles = DEFAULT_ROLES
+    iam_roles = get_iam_roles_from_file(user)
```

Comment:
or maybe you could wrap this in a try/except, and have the except set the roles to `DEFAULT_ROLES` and log the failure reason (invalid json, missing file, etc)